bookwyrm/migrations/0151_user_show_guided_tour.py

```diff
@@ -0,0 +1,19 @@
+# Generated by Django 3.2.13 on 2022-06-13 01:59
+
+import bookwyrm.models.fields
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("bookwyrm", "0150_readthrough_stopped_date"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="user",
+            name="show_guided_tour",
+            field=bookwyrm.models.fields.BooleanField(default=True),
+        ),
```

Comment:
I think for existing users, we don't want to show the tour (right?) -- the default value is correct here, but there could be a `RunPython` step that sets the boolean to false for existing users.

Reply:
That is probably sensible. I'm bit sure how to do that - can you point me to the relevant script/file?

Reply:
There are a bunch in the migrations, this being one: https://github.com/bookwyrm-social/bookwyrm/blob/main/bookwyrm/migrations/0142_auto_20220227_1752.py#L7. But the docs are actually probably clearer than the examples: https://docs.djangoproject.com/en/3.2/ref/migration-operations/#django.db.migrations.operations.RunPython

Reply:
I'm having a bit of trouble understanding the docs. Does `RunPython` run immediately after the migration? The way it's written sound to me like it runs immediately before, which doesn't seem all that helpful.

Would I add a function to `bulk_update` here to overwrite the default for existing users?

Reply:
The migration runs a series of steps that are in the `operations` array, and since they're in an array, they run sequentially. That means you can add your own step before or after any of the steps in the migrations. If you had a bunch of different fields changed, each `AddField`, `RemoveField`, `AlterField`, or whatever, would be its own step in the `operations`, and you could add `RunPython` scripts anywhere you'd like.

 So you have:
``` python
    operations = [
        migrations.AddField(
            model_name="user",
            name="show_guided_tour",
            field=bookwyrm.models.fields.BooleanField(default=True),
        ),
    ]
```

And you can add a script step after `AddField`:
``` python
    operations = [
        migrations.AddField(
            model_name="user",
            name="show_guided_tour",
            field=bookwyrm.models.fields.BooleanField(default=True),
        ),
        migrations.RunPython(forward_function, migrations.RunPython.noop),
    ]
```

Then you just have to write a function (which here I called `foreward_function` to be generic) that does what you want. The second parameter (I used `migrations.RunPython.noop`, which does nothing) runs if you undo the migration. 

I don't know why, but in a migration function you have to call models using `schema_editor` and `db_alias`, so what would normally be:
``` Python
from bookwyrm import models
def forward_func():
    models.User.objects.filter(...)
```

Becomes
``` Python
def forward_func(apps, schema_editor):
    db_alias = schema_editor.connection.alias
    user_model = apps.get_model("bookwyrm", "User")
    user_model.objects.using(db_alias).filter(...)
```

I think you can use `.update()` on the queryset to bulk update? Something like:
``` Python
user_model.objects.using(db_alias).filter(local=True).update(show_guided_tour=False)
```

But there might be a more efficient way to do that.